src/vcpkg/install.cpp

```diff
@@ -1084,12 +1084,23 @@ namespace vcpkg
             {
                 features.emplace_back("core");
             }
-
+            PackageSpec toplevel{manifest_core.name, default_triplet};
             auto core_it = std::remove(features.begin(), features.end(), "core");
             if (core_it == features.end())
             {
-                const auto& default_features = manifest_core.default_features;
-                features.insert(features.end(), default_features.begin(), default_features.end());
+                if (Util::any_of(manifest_core.default_features, [](const auto& f) { return !f.platform.is_empty(); }))
+                {
+                    const auto& vars = var_provider.get_or_load_dep_info_vars(toplevel, host_triplet);
+                    for (const auto& f : manifest_core.default_features)
+                    {
+                        if (f.platform.evaluate(vars)) features.push_back(f.name);
+                    }
+                }
+                else
+                {
+                    for (const auto& f : manifest_core.default_features)
+                        features.push_back(f.name);
+                }
```

Comment:
Question unrelated to the PR: 
Does this change the timing when the platform expressions gets evaluated? When I tried to implement `"system": "<platformexpression>"` in the ports manifest i saw that the evaluation was one step late. Meaning that the port and all its deps were already added to the graph and I couldn't stop that without making a bigger change (and not really knowing where to start ue to time constraints.).

Reply:
> Does this change the timing when the platform expressions gets evaluated?

No